vNext/extensions/applicationinsights-react-native/src/ReactNativePlugin.ts

```diff
@@ -0,0 +1,106 @@
+/**
+ * ReactNativePlugin.ts
+ * @copyright Microsoft 2019
+ */
+
+import {
+    ITelemetryPlugin,
+    ITelemetryItem,
+    IPlugin,
+    IConfiguration,
+    IAppInsightsCore
+} from '@microsoft/applicationinsights-core-js';
+import { ConfigurationManager, DeviceExtensionKeys } from '@microsoft/applicationinsights-common';
+import DeviceInfo from 'react-native-device-info';
+
+import { INativeDevice, IReactNativePluginConfig } from './Interfaces';
+
+export class ReactNativePlugin implements ITelemetryPlugin {
+    private _initialized: boolean = false;
+    private _device: INativeDevice;
+    private _config: IReactNativePluginConfig;
+
+    identifier: string = 'AppInsightsReactNativePlugin';
+    priority: number = 199;
+    _nextPlugin?: ITelemetryPlugin;
+
+    constructor(config?: IReactNativePluginConfig) {
+        this._config = config || this._getDefaultConfig();
+        this._device = {};
+    }
+
+    public initialize(
+        config?: IReactNativePluginConfig | object, // need `| object` to coerce to interface
```

Comment:
shouldnt this be IConfiguration | IConfig

Reply:
Or just IConfiguration, and extensions.AppInsightsReactNativePlugin = {} will be individual config. Lets try to dissuade from setting up root level global configs if possible.

Reply:
@jpiyali - I want the way to modify config options here to be from the constructor, but am wondering if I still need to bother supporting `extensionsConfig` as well. If not, I can remove all of the `ConfigurationManager` logic here